homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -75,51 +81,85 @@ async def async_setup_entry(
 class YaleAlarmDevice(CoordinatorEntity, AlarmControlPanelEntity):
     """Represent a Yale Smart Alarm."""
 
+    _attr_code_arm_required = False
+    _attr_supported_features = SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY
+
     def __init__(self, coordinator: YaleDataUpdateCoordinator) -> None:
         """Initialize the Yale Alarm Device."""
         super().__init__(coordinator)
-        self._attr_name: str = coordinator.entry.data[CONF_NAME]
+        self._coordinator = coordinator
+        self._attr_name = coordinator.entry.data[CONF_NAME]
         self._attr_unique_id = coordinator.entry.entry_id
-        self._identifier: str = coordinator.entry.data[CONF_USERNAME]
-
-    @property
-    def device_info(self) -> DeviceInfo:
-        """Return device information about this entity."""
-        return DeviceInfo(
-            identifiers={(DOMAIN, self._identifier)},
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.entry.data[CONF_USERNAME])},
             manufacturer=MANUFACTURER,
             model=MODEL,
-            name=str(self.name),
+            name=self._attr_name,
         )
 
-    @property
-    def state(self):
-        """Return the state of the device."""
-        return STATE_MAP.get(self.coordinator.data["alarm"])
-
-    @property
-    def available(self):
-        """Return if entity is available."""
-        return STATE_MAP.get(self.coordinator.data["alarm"]) is not None
-
-    @property
-    def code_arm_required(self):
-        """Whether the code is required for arm actions."""
-        return False
-
-    @property
-    def supported_features(self) -> int:
-        """Return the list of supported features."""
-        return SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY
-
-    def alarm_disarm(self, code=None):
+    async def async_alarm_disarm(self, code=None) -> None:
         """Send disarm command."""
-        self.coordinator.yale.disarm()
-
-    def alarm_arm_home(self, code=None):
+        if self._coordinator.yale:
+            try:
+                alarm_state = await self.hass.async_add_executor_job(
+                    self._coordinator.yale.disarm
+                )
+            except (
+                AuthenticationError,
+                ConnectionError,
+                TimeoutError,
+                UnknownError,
+            ) as error:
+                LOGGER.warning("Could not verify disarmed: %s", error)
+
+        if alarm_state:
```

Comment:
Is there other data worth refreshing as a result of this RPC? In particular in the error case -- if there was a timeout, maybe it actually succeeded on the server side but the response was dropped.  The example in https://developers.home-assistant.io/docs/integration_fetching_data?_highlight=dataupdatecoordinator#coordinated-single-api-poll-for-data-for-all-entities shows another fetch after performing a write to get the device view of the state, so its something you could consider here.

Reply:
The refresh is only 15 seconds so I skipped it but probably you're right so I will put it in

Reply:
This does not work. The api doesn't record the status fast enough in the backend so if I press alarm_home then it sets it and then reverts back to disarmed until a second refresh is done. No point in adding a delay so I will keep it as original.